src/com/google/enterprise/adaptor/fs/WindowsFileDelegate.java

```diff
@@ -538,19 +581,26 @@ public void callback(int errorCode, int nBytesTransferred,
 
         // Signal any waiting threads that the monitor is now active.
         startSignal.countDown();
+        backOff.reset();
 
         boolean logging = !paused();
         if (logging) {
           log.log(Level.FINER, "Waiting for notifications for {0}.", watchPath);
         }
         int waitResult = kernel32.WaitForSingleObjectEx(stopEvent,
-            Kernel32.INFINITE, true);
+            15 * 60 * 1000 /* 15 min timeout in millisecs */, true);
         if (waitResult == Kernel32Ex.WAIT_IO_COMPLETION) {
           if (logging) {
             log.log(Level.FINER, "A notification was sent to the monitor "
```

Comment:
You said we get an empty change list when the server goes down. What happens here and in the callback (which has a do-while loop)?